great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -94,17 +112,35 @@ def __init__(self, *args, **kwargs):
             self._s3 = None
 
         try:
-            self._azure = BlobServiceClient(**azure_options)
+            if "conn_str" in azure_options:
+                self._azure = BlobServiceClient.from_connection_string(**azure_options)
+            else:
+                self._azure = BlobServiceClient(**azure_options)
```

Comment:
Missed the alternative constructor on my first implementation of this! These `try/except` clauses should mirror what happens in the `DataConnector` constructors